actions/Network.js

```diff
@@ -31,14 +28,8 @@ function createWretchError(status, message) {
 function wretchWithOptions(url, shouldRetry = false) {
   return wretch(url)
     .middlewares(shouldRetry ? [retry()] : [])
-    .addon(AbortAddon())
-    .resolve((resolver) => resolver.setTimeout(REQUEST_TIMEOUT))
```

Comment:
@vtsaplin , Could you please elaborate on the benefits of setting a timeout here? If a request doesn't finish before hitting AppBuilder execution timeout limits, the action will fail regardless. The end service will still process the request, and it won't be aborted by simply using a timeout on our side. Additionally, according to the [documentation](https://github.com/elbywan/wretch?tab=readme-ov-file#abort-), AbortAddon() is intended for use by web clients only, not on the server side.

Reply:
@slitviachenko The goal is to gracefully stop the action, return a meaningful error message, and clearly explain the reason for the failure. This is important because these messages are shown to users. Without this, abrupt disconnects are managed by a gateway, which sends back a very unfriendly error message (http 503, if I remember correctly). This type of message isn't just unhelpful, it also makes it hard to tell apart from an actual crash.

Reply:
`wretch` is isomorphic, and I assume that applies to its addons too. The `AbortController` is part of the NodeJS API. Here are the correct docs: https://nodejs.org/api/globals.html#class-abortcontroller

Reply:
>which sends back a very unfriendly error message (http 503, if I remember correctly)

Yes, I checked and found that, unfortunately, AppBuilder defaults to returning a 503 text/html error :(
```
< HTTP/2 503
< content-type: text/html
...
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
<TITLE>ERROR: The request could not be satisfied</TITLE>
</HEAD><BODY>
<H1>503 ERROR</H1>
<H2>The request could not be satisfied.</H2>
<HR noshade size="1px">
The Lambda function associated with the CloudFront distribution is invalid or doesn't have the required permissions.
We can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.
<BR clear="all">
If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.
<BR clear="all">
<HR noshade size="1px">
<PRE>
Generated by cloudfront (CloudFront)
Request ID: DdcOlaN3gu9cP0doBByORiX2s1BO27HaVzybxM-QIadnOrMuXMJQiQ==

...
```

Reply:
@vtsaplin ,
>return a meaningful error message, and clearly explain the reason for the failure
BTW, a 400 error is currently being returned for all Firefall errors due to https://github.com/adobe/aem-genai-assistant/blob/main/actions/FirefallClient.js#L28
```
return new InternalError(400, `IS-ERROR: ${errorMessage} (${error.status}).`);
```

I fixed that and made it possible to configure the timeout. Please review my latest changes when you have a chance.

CC: @askayastha22

Reply:
@vtsaplin ,
>wretch is isomorphic, and I assume that applies to its addons too

I haven't checked, but I hope so. Anyway the end service will continue to process the request, and it won't be interrupted by a timeout on our side. :(